doc/src/vtr/server_mode/index.rst

```diff
@@ -0,0 +1,203 @@
+.. _server_mode:
+
+Server Mode
+================
+
+VTR provides the ability to run in server mode using the following command-line arguments.
+
+.. code-block:: none
+
+  --server --port_num 60555
+
+Server mode may only accept a single client application connection and respond to two types of requests: **get critical path report** and **highlight selected critical path elements**.
+
+Communication telegram
+-------------------------
+
+Telegram consists of two parts: a fixed-size **telegram header** and a variable-size **telegram body**.
+The telegram header contains helper information required to properly extract the telegram body sequence from the data flow.
+
+.. _fig_comm_telegram_structure:
+
+.. figure:: comm_telegram_structure.*
+
+    Communication telegram structure.
+
+    
+.. _fig_comm_telegram_body_structure:
+
+.. figure:: comm_telegram_body_structure.*
+
+    Communication telegram body structure.
+
+    **telegram body** is flat **JSON** structure.
+
+    **CMD** could have following integer values:
+
+    - 0 - command id for **get critical path**
+    - 1 - command id for **highlight selected path elements**
+
+    JOB_ID - is unique id for a task.
```

Comment:
Explain how JOB_ID should be set and its purpose.  Any 32-bit integer?  Should each be unique in a session, or just different from any outstanding request?  Is the JOB_ID sent in a command in the same JOB_ID contained in a response so you can identify which request the response matches?

Reply:
"command in the same JOB_ID contained in a response so you can identify which request the response matches"
yes this is purpose of JOB_ID.
solved 
https://github.com/w0lek/vtr-verilog-to-routing/pull/11/commits/5d186029dcafd55b5a4365397732bc7daf199065
(will be updated later)